File: Find First and Last Position of Element in Sorted Array.cpp
#include<vector>
#include<iostream>

using namespace std;

class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int left=binSearch(nums,target);
        int right=binSearch(nums,target+1)-1;
        if(left <=right){
            return {left,right};
        }

        return {-1,-1};
        
    }
public:
    int binSearch(vector<int>& nums, int target){
        int n=nums.size();
        int first_pos=n;
        int low = 0,high = n-1;

        while(low <= high){
            int mid=low+(high-low)/2;
            if  (nums[mid] >=target){
                first_pos=mid;
                high = mid-1;

            }else{
                low = mid+1;
            }
        }
        return first_pos;

    };
};

int main(){
    Solution s;

    int target;
    vector<int> nums;




    s.searchRange(nums,target);


    return 0;
}

File: Find First and Last Position of Element in Sorted Array.py
class Solution:
    def searchRange(self, nums, target):
        left=self.binSearch(nums,target,True)
        right=self.binSearch(nums,target,False)

        return [left,right]



    def binSearch(self,nums,target,leftBias):

        l,r=0,len(nums)-1
        i=-1
        while (l<=r):   
            m=(l+r)//2
            if target > nums[m]:
                l=m+1
            elif target < nums[m]:
                r=m-1
            else:
                i=m
                if leftBias:
                    r=m-1
                else:
                    l=m+1
        return i

        
        
        



nums = [5,7,7,8,8,10]
target = 8
# nums = [5,7,7,8,8,10]
# target = 6
# nums = []
# target = 0
sl=Solution()
print(sl.searchRange(nums,target))
